src/regdesk/states/multiple_result_state.js

```diff
@@ -0,0 +1,55 @@
+import { RegState } from './reg_state.js';
+
+/**
+ * Multiple result state
+ */
+export class MultipleResultState extends RegState {
+  /**
+   * Get the transition events this state can process.
+   */
+  static get events() {
```

Comment:
I am just now realizing you define these events a state is responsible for twice. Once inside the state and once inside the mgr. Can you reduce that down to just one? I think having it contained in just the state is best.

Reply:
This static getter here is where we define the events that a state can handle, while what we define in the mgr class is the relationship between a state's events and the _next_ state we end up as a result of processing that event.

I played with this back and forth a bit, initially states _did_ have an awareness of each other, so the LoadingState would have its own internal map stating that when a SINGLE_RESULT event fired it would lead to the SingleResultState. I moved away from this because I felt like it was coupling the states together in a way that made the individual states need too much information about other states.

In this way, states just fire off events that need to be handled elsewhere, and detail what those events could be. It's up to the manager to handle mapping those events to the appropriate states.

tldr: The mapping should be in the manager so we can modify it on the fly. The definitions should be in the state class.

Reply:
Half of me is frustrated that it requires modifying two different files/places when changing state code. But I get what you're saying and I do agree with you. So for now this is probably fine, I'll keep an eye on it. :)

Reply:
It's definitely walking the line of the SRP/SoC guidance, it's fair feedback.